backend/msgpack/queue.go

```diff
@@ -120,21 +125,43 @@ func (q *queue) drain() {
 				xlog.NewLogField("instance", q.instance.Endpoint()),
 				xlog.NewLogErrField(err),
 			).Error("write data error")
-			q.metrics.connWriteErrors.Inc(1)
 		}
 		buf.Close()
 	}
 	q.conn.Close()
 }
 
+func (q *queue) reportQueueSize(reportInterval time.Duration) {
+	ticker := time.NewTicker(reportInterval)
+	for range ticker.C {
+		q.RLock()
```

Comment:
Talked offline, clarified that we only want to use the done channel in `reportQueueSize` so that we don't need to acquire any locks which may impede metric ingestion.